src/PublicVault.sol

```diff
@@ -489,11 +489,18 @@ contract PublicVault is VaultImplementation, IPublicVault, ERC4626Cloned {
         );
 
       VaultData storage s = _loadStorageSlot();
+
       if (amount > s.balance) {
         revert InvalidVaultState(
           InvalidVaultStates.LOAN_GREATER_THAN_VIRTUAL_BALANCE
         );
       }
+
+      if (s.balance - amount < s.withdrawReserve) {
+        revert InvalidVaultState(
+          InvalidVaultStates.WITHDRAW_RESERVE_UNDER_COLLATERALIZED
+        );
+      }
```

Comment:
the first conditional will always be true if the second one is i think so can delete first unless the error difference matters

Reply:
I went back and forth on this one, but I concur. We need two different errors